test/Poly.jl

```diff
@@ -421,7 +421,25 @@ fit(Poly, xx,yy,2)
 ## Issue with overflow and polyder Issue #159
 @test !iszero(polyder(Poly(BigInt[0, 1])^100, 100))
 
+@testset "`isreal` and `real`" begin
+    x = Polynomial([1 // 2, 2 + 0im, 3.0, 4.0 + 0.0im])
+    y = Polynomial([1 // 2, 2 + 0im, 3.0, 4.0 + 0.1im])
+    @test isreal(x) === true
+    @test isequal(x, real(x)) === true
+    @test eltype(real(x)) === Float64
+    @test real(x) == Polynomial([1 // 2, 2, 3, 4.0])
+    @test isreal(y) === false
+    @test real(y) == real(x)
+end
 
+@testset "`isintegerpoly`" begin
+    x = Polynomial([1 // 1, Int8(2) + 0im, 3.0, Int16(4) + 0im])
+    y = Polynomial([1 // 2, Int8(2) + 0im, 3.0, Int16(4) + 0im])
+    @test isintegerpoly(x) === true
+    @test isintegerpoly(y) === false
+    @test convert(Polynomial{Int}, x) == Polynomial([1, 2, 3, 4])
```

Comment:
One thing I don't understand: `convert(Polynomial{Int}, x) == Polynomial([1, 2, 3, 4])` is `true` but `convert(Polynomial{Int}, x) === Polynomial([1, 2, 3, 4])` is `false`. Is this because the difference between `x.coeffs` or `x.var`?

Reply:
It is because `[1,2,3,4] === [1,2,3,4]` is `false`.

Reply:
One thing I've thought of is the `as_` function. The use of "as" is from my R experience, where "as" is used for conversion. More broadly though this pattern:

```
asXXX(fn, p) = ⟒(typeof(p))(fn.(coeffs(p)), p.var)
```

(or some variant) is really just broadcasting `fn` over the coefficients then reassembling. Really we should be calling it broadcast or something similar. Maybe `broadcast_function` or `map_function` (the latter as it doesn't handle broadcasting over all variables...). Even if `convert(Polynomial{Int}, x)` is the most idiomatic way to convert to a polynomial with coefficients of type `Int`, this would be a useful utility function to have.

Reply:
In looking at the docstring for `map`, this seems to be in line with its generic meaning, viewing a polynomial as some sort of container for its coefficients. Perhaps:

```
 Base.map(f, p::P) where {P <: AbstractPolynomial} =⟒(P)(map(f, coeffs(p)), p.var)
```

is the most idiomatic way to handle this "as" functionality. Though, the case of `SparsePolynomials` might need a special method, as the `coeffs(p)` call can be costly.